spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/RetryableRibbonLoadBalancingHttpClient.java

```diff
@@ -124,14 +125,17 @@ public RibbonApacheHttpResponse doWithRetry(RetryContext context) throws IOExcep
 				RibbonApacheHttpRequest newRequest = request;
 				if(context instanceof LoadBalancedRetryContext) {
 					ServiceInstance service = ((LoadBalancedRetryContext)context).getServiceInstance();
-					if(service != null) {
-						//Reconstruct the request URI using the host and port set in the retry context
-						newRequest = newRequest.withNewUri(UriComponentsBuilder.newInstance().host(service.getHost())
-								.scheme(service.getUri().getScheme()).userInfo(newRequest.getURI().getUserInfo())
-								.port(service.getPort()).path(newRequest.getURI().getPath())
-								.query(newRequest.getURI().getQuery()).fragment(newRequest.getURI().getFragment())
-								.build().encode().toUri());
+					try {
```

Comment:
Why do you catch the exception in the Apache implementation and not in the OK implementation?

Reply:
The `RetryCallback` within the Apache implementation is defined for an `IOException` where the OK implementation is less specific and handles an `Exception`.   Updating the Apache implementation so the callback is less specific would improve this change.

Reply:
Hello @ryanjbaxter.  There are a few unused imports in both the Apache and OK retryable clients.  Would you mind if I pushed another update?